src/agent/onefuzz/src/syncdir.rs

```diff
@@ -305,6 +330,29 @@ impl SyncedDir {
 
                 event!(event.clone(); EventData::Path = file_name_event_str);
```

Comment:
Have you looked into hooking these `send_direct` calls right into the `event!` or `metric!` macros?

Reply:
Yes. I'd tried and ran into a few cyclic dependencies that I was having trouble getting around. 

More importantly, going forward I don't think metrics will always coincide with send_direct calls. Currently, it just so happens that we are only making metric calls when reporting job data, but I see a future where a metric is used more broadly.